source/opt/dominator_tree.cpp

```diff
@@ -227,23 +227,28 @@ bool DominatorTree::StrictlyDominates(const ir::BasicBlock* a,
   return DominatorTree::StrictlyDominates(a->id(), b->id());
 }
 
+bool DominatorTree::StrictlyDominates(const DominatorTreeNode* a,
+                                      const DominatorTreeNode* b) const {
+  if (a == b) return false;
+  return Dominates(a, b);
+}
+
 bool DominatorTree::Dominates(uint32_t a, uint32_t b) const {
   // Check that both of the inputs are actual nodes.
-  auto a_itr = nodes_.find(a);
-  auto b_itr = nodes_.find(b);
-  if (a_itr == nodes_.end() || b_itr == nodes_.end()) return false;
+  const DominatorTreeNode* a_node = (*this)[a];
```

Comment:
This will create entries (with nullptr) for IDs that are not present.  That's undesirable.  The old behaviour was better.

Reply:
I overloaded the operator (that's calling DominatorTree::operator[]). If a is not in the tree, the returned pointer is null.